radicle/src/git.rs

```diff
@@ -197,22 +204,76 @@ pub fn write_tree<'r>(
 
 /// Configure a repository's radicle remote.
 ///
-/// Takes the repository in which to configure the remote, the name of the remote, the public
-/// key of the remote, and the path to the remote repository on the filesystem.
+/// The entry for this remote will be:
+/// ```text
+/// [remote.<name>]
+///   url = <url>
+///   fetch +refs/heads/*:refs/remotes/<name>/*
+/// ```
 pub fn configure_remote<'r>(
     repo: &'r git2::Repository,
-    remote_name: &str,
-    remote_id: &RemoteId,
-    remote_url: &Url,
+    name: &str,
+    url: &Url,
 ) -> Result<git2::Remote<'r>, git2::Error> {
-    let fetch = format!("+refs/remotes/{remote_id}/heads/*:refs/remotes/rad/*");
-    let push = format!("refs/heads/*:refs/remotes/{remote_id}/heads/*");
-    let remote = repo.remote_with_fetch(remote_name, remote_url.to_string().as_str(), &fetch)?;
-    repo.remote_add_push(remote_name, &push)?;
+    let fetch = format!("+refs/heads/*:refs/remotes/{name}/*");
+    let remote = repo.remote_with_fetch(name, url.to_string().as_str(), &fetch)?;
 
     Ok(remote)
 }
 
+// TODO: Would be nice if libgit2 respected namespaces https://github.com/libgit2/libgit2/issues/6414
+/// Fetch from the given `remote` using the provided `namespace`.
+///
+/// This uses [`Command`] under the hood and is the equivalent to:
+///
+///  `GIT_NAMESPACE=<namespace> git fetch <remote>`
+pub fn fetch(repo: &git2::Repository, remote: &str, namespace: &RemoteId) -> io::Result<()> {
+    let out = Command::new("git")
+        .current_dir(repo.path())
+        .env("GIT_NAMESPACE", Component::from(namespace).as_str())
+        .args(["fetch", remote])
+        .output()?;
+    log::trace!(
+        "stdout: {}, stderr: {}",
+        String::from_utf8_lossy(&out.stdout),
+        String::from_utf8_lossy(&out.stderr)
+    );
+    Ok(())
+}
+
+// TODO: Would be nice if libgit2 respected namespaces https://github.com/libgit2/libgit2/issues/6414
+// TODO: could pass refspecs for pushing
```

Comment:
I think this is a good idea, I'm not a fan of optional parameters in rust using `Option`; we could take a slice of pushspecs instead :shrug: